packages/next/src/server/app-render/encryption.ts

```diff
@@ -81,17 +85,29 @@ async function encodeActionBoundArg(actionId: string, arg: string) {
 export async function encryptActionBoundArgs(actionId: string, args: any[]) {
   const { clientModules } = getClientReferenceManifestForRsc()
 
-  // Create an error before any asynchrounous calls, to capture the original
+  // Create an error before any asynchronous calls, to capture the original
   // call stack in case we need it when the serialization errors.
   const error = new Error()
   Error.captureStackTrace(error, encryptActionBoundArgs)
 
   let didCatchError = false
 
+  const workUnitStore = workUnitAsyncStorage.getStore()
+
+  const hangingInputAbortSignal =
+    workUnitStore?.type === 'prerender'
+      ? createHangingInputAbortSignal(workUnitStore)
+      : undefined
+
   // Using Flight to serialize the args into a string.
   const serialized = await streamToString(
     renderToReadableStream(args, clientModules, {
```

Comment:
Can this code ever run in edge runtime? the hanging input abort signal uses node only APIs so we may need an edge implementation too

Reply:
We're only creating the hanging input abort signal if Dynamic I/O is enabled, but Dynamic I/O is not supported anymore in the edge runtime since #71500. Was this maybe a miscommunication and unintentional?

Reply:
I guess it won’t ever call the function but I think we have a transform in SWC that warns if you use process in edge bundles and I think this might run into that problem

Reply:
Yeah, that sounds like it should indeed happen. Interesting that no tests caught that, but probably because we don't assert on no warnings being emitted.

Reply:
I just tested it with `test/e2e/app-dir/actions` which has an inline server action with bound arguments using the edge runtime. Here's what I found:
- Generally we do emit a build error, not a warning if incompatible edge APIs are found.
- In this case no error es emitted, most likely due to an overzealous relaxation in #69283.
- If I move the `process.nextTick` usage out of the function and into the module scope, a build error is emitted.

I now moved the function back to `packages/next/src/server/app-render/dynamic-rendering.ts`, and switched to using `scheduleOnNextTick`, which is compatible with both runtimes, and avoids build errors due to the `process.env.NEXT_RUNTIME` guard.